config.go

```diff
@@ -219,9 +217,19 @@ func (y *YAML) withDefault(d interface{}) (*YAML, error) {
 		return nil, fmt.Errorf("merging default and existing YAML failed: %v", err)
 	}
 
+	// Since we've gone back to working with the original input sources, we also
+	// need to re-expand environment variables. Since we only need to do this to
+	// support this deprecated method, keep things simple and reach out into the
+	// environment again.
+	merged, err = expandVariables(y.lookup, merged)
```

Comment:
this looks fine, but a lot of this logic looks very similar to `NewYAML`. Can we instead use `NewYAML` (maybe we provide an unexported option that lets you add `[]config.source` directly). I think the code could instead be:

```
opts := []yaml.YAMLOption{
  Name(y.Name),
  Expand(y.lookup),
  Source(rawDefault),
  rawSources(y.raw),
}
if !y.strict {
  opts = append(opts, Permissive())
}

return NewYAML(opts...)
```

The "merging default and existing YAML" error message might not be as specific to default, but it seems like less code to maintain, and less likely to run into problems like this.

Reply:
done